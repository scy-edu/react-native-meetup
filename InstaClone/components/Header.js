'use strict';

import React, {
  Component,
} from 'react';
import {
	View,
  Text,
  StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

export const styles = StyleSheet.create({
  default: {
		// Dimensions
		height: 80,
		width: '100%',

		// Border bottom colors
		borderBottomWidth: 0.2,
		borderBottomColor: 'grey',

		// padding
		paddingBottom: 5,

		// align items
		justifyContent: 'flex-end',
	},
	title: {
		alignSelf: 'center',
		fontFamily: 'BradleyHandITCTT-Bold',
		fontSize: 35,
	},
});

const Header = ({
  name,
  type,
}) => (
  <View style={styles[type]}>
    <Text style={styles['title']}>{name}</Text>
  </View>
);

Header.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

Header.defaultProps = {
  name: 'InstaClone',
  type: 'default',
};

Header.displayName = 'Header';

export default Header;
