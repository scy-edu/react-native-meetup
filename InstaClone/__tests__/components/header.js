'use strict';

import 'react-native';
import React from 'react';
import Header, {
	styles,
} from '../../components/Header';

import renderer from 'react-test-renderer';
import {
	shallow,
} from 'enzyme';

const createTestProps = () => ({
	name: 'InstaClone',
	type: 'default',
});

describe('<Header />', () => {
	let wrapper;

	describe('rendering', () => {
		// run before each test
		beforeEach(() => {
			wrapper = shallow(<Header {...createTestProps()} />);
		});

		it('renders correctly', () => {
			const tree = renderer.create(<Header />).toJSON();
			expect(tree).toMatchSnapshot();
		});

		it('renders InstaClone in the name', () => {
			expect(wrapper.find('Text').contains('InstaClone')).toBe(true);
		});

		describe('styling', () => {
			it('should have the title style', () => {
				expect(wrapper.find('Text').prop('style')).toBe(styles['title']);
			});
		});
	});

});
