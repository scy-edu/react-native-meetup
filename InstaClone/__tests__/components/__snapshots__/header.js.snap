// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> rendering renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "grey",
      "borderBottomWidth": 0.2,
      "height": 80,
      "justifyContent": "flex-end",
      "paddingBottom": 5,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "BradleyHandITCTT-Bold",
        "fontSize": 35,
      }
    }
  >
    InstaClone
  </Text>
</View>
`;

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "grey",
      "borderBottomWidth": 0.2,
      "height": 80,
      "justifyContent": "flex-end",
      "paddingBottom": 5,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "BradleyHandITCTT-Bold",
        "fontSize": 35,
      }
    }
  >
    InstaClone
  </Text>
</View>
`;
